### PR TITLE
Fix async code calling blocking sync code

### PR DESCRIPTION
**Pull request by dekkers**
### Changes

This fixes async functions calling blocking sync code. After spotting this in a code review, I checked and saw that we already did this incorrectly in a few other places. Async code should not call blocking sync code because that will block the whole async event loop.

### QA notes

It is hard to verify that it was not correct before, because the only effect is that the code would block in a place where it shouldn't.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/octopoes/api/router.py [hunk 1]
```diff
@@ -568,18 +568,18 @@ def importer(data: bytes, xtdb_session_: XTDBSession, reset: bool = False) -> di
...
         raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="Error receiving objects") from e
     return importer(data, xtdb_session_)
 
 
 @router.post("/io/import/new", tags=["io"])
-async def importer_new(request: Request, xtdb_session_: XTDBSession = Depends(xtdb_session)) -> dict[str, int]:
+def importer_new(request: Request, xtdb_session_: XTDBSession = Depends(xtdb_session)) -> dict[str, int]:
     try:
-        data = await request.body()
+        data = request.body()
```
Comment (Donnype) [inline octopoes/octopoes/api/router.py:582]: Here we do call async code though?

File: bytes/bytes/api/router.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ async def create_raw(
         if meta_repository.has_raw(meta, parsed_mime_types):
             return RawResponse(status="success", message="Raw data already present")
 
-        data = await request.body()
+        data = request.body()
```
Comment (Donnype) [inline bytes/bytes/api/router.py:164]: Same here?

File: boefjes/boefjes/api.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ async def boefje_input(
 
 
 @app.post("/api/v0/tasks/{task_id}")
-async def boefje_output(
+def boefje_output(
```
Comment (Donnype) [inline boefjes/boefjes/api.py:105]: :+1:

File: octopoes/octopoes/api/router.py (hunk 1 above)
Comment (dekkers) [reply octopoes/octopoes/api/router.py]: Yes, so I need to fix it differently. I've added `sync_to_async` decorator to the importer so we can await the importer. Even if you call async code, you can't just mix sync and async code.

File: bytes/bytes/api/router.py (hunk 2 above)
Comment (dekkers) [reply bytes/bytes/api/router.py]: I have wrapped this with `async_to_sync` because all the other code in the function also calls sync code.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Onboarding works, scanning works, findings are created.

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**